### ISSUE TITLE
Provide trivial inferencing/reasoning-like support for membership in namespaces

### ISSUE THREAD

**Issue by rlaemmel** [101companies/101wiki#17]
Namespaces provided a non-semantic means of classification.

For instance, by using the prefix "Language" in "Language:Haskell", it is meant to be clear that "Language:Haskell" is an instance of "Namespace:Language". 

However, at the triple level, this fact needs to be declared.

The new 101wiki should probably auto-inject properties like

[[instanceOf::Namespace:Language]].

For the time being, this sort of property is manually added, whenever the wiki user things of this issue.

Probably, we need to discuss this some more before we really go into this.

**Comment by tschmorleiz** [comment]
One cheap thing we could do is add [[instanceOf::Namespace:X]] whenever X:Y is changed and does not have that metadata. I think that would be ok for now.

**Comment by rlaemmel** [comment]
... with special handling for "@" and "Concept".
Otherwise, this would be great.
Will this work all fine when thinks are moved across namespaces?

**Comment by tschmorleiz** [comment]
Will work, I will implement it in a "change" method which is called both on saving and renaming.

**Comment by tschmorleiz** [comment]
Now on every saving and and renaming the namespace triple is injected. When renaming such that the namespace is changed, the triple is updated.

We still have a lot of pages not having the triple. Adding will be triggered by saving, but I guess I will do this with some script which checks all pages.

**Comment by rlaemmel** [comment]
How can I test this?

I was not able to get what I expected.

Here is what I did.
- I went here: http://101companies.org/wiki/Function
- Did some changes and saved and even reloaded.
- No instanceOf::Namespace:Concept appear, neither in page source, nor in rendering.
- Just for fun, I added it manually and it appeared as expected.
- Removed it again (manually).

Also tried this here:
- http://101companies.org/wiki/@technology (added instanceOf manually)
- http://101companies.org/wiki/Technology:HsFragmentLocator (Haven't yet added instanceOf)
